src/PackageManagement.VisualStudio/IDE/VSDeleteOnRestartManager.cs

```diff
@@ -0,0 +1,233 @@
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Microsoft.VisualStudio.Shell;
+using NuGet.Configuration;
+using NuGet.Packaging.Core;
+using NuGet.ProjectManagement;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// An <see cref="DeleteOnRestartManager"/> manger which is used for surfacing errors and UI in VS.
+    /// </summary>
+    [Export(typeof(IDeleteOnRestartManager))]
+    public class VsDeleteOnRestartManager : IDeleteOnRestartManager
+    {
+        // The file extension to add to the empty files which will be placed adjacent to partially uninstalled package
+        // directories marking them for removal the next time the solution is opened.
+        private const string DeletionMarkerSuffix = ".deleteme";
+        private const string DeletionMarkerFilter = "*" + DeletionMarkerSuffix;
+
+        private string _packagesFolderPath = null;
+
+        public event EventHandler<PackagesMarkedForDeletionEventArgs> PackagesMarkedForDeletionFound;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="VsDeleteOnRestartManager"/>.
+        /// </summary>
+        public VsDeleteOnRestartManager() : this(
+            ServiceLocator.GetInstance<ISettings>(),
+            ServiceLocator.GetInstance<ISolutionManager>())
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="DeleteOnRestartManager"/>.
+        /// </summary>
+        /// <param name="settings">The <see cref="ISettings"/> associated with the current solution.</param>
+        /// <param name="solutionManager">The <see cref="ISolutionManager"/> associated with the current solution.
+        /// </param>
+        public VsDeleteOnRestartManager(
+            ISettings settings,
+            ISolutionManager solutionManager)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            if (solutionManager == null)
+            {
+                throw new ArgumentNullException(nameof(solutionManager));
+            }
+
+            Settings = settings;
+            SolutionManager = solutionManager;
+            SolutionManager.SolutionOpened += OnSolutionOpenedOrClosed;
+            SolutionManager.SolutionClosed += OnSolutionOpenedOrClosed;
+        }
+
+        public ISolutionManager SolutionManager { get; }
+
+        public ISettings Settings { get; }
+
+        public string PackagesFolderPath
+        {
+            get
+            {
+                if (SolutionManager.SolutionDirectory != null)
+                {
+                    _packagesFolderPath =
+                        PackagesFolderPathUtility.GetPackagesFolderPath(SolutionManager, Settings);
+                }
+
+                return _packagesFolderPath;
+            }
+
+            set
```

Comment:
I don't think this setter or the local '_packagesFolderPath' is valid anymore. I would definitely make this getter only property and do the following

+                if (SolutionManager.SolutionDirectory != null)
+                {
+                    return
+                        PackagesFolderPathUtility.GetPackagesFolderPath(SolutionManager, Settings);
+                }
+
+                return null;